src/zoo.js

```diff
@@ -1,79 +1,241 @@
 const data = require('./data')
 
 function entryCalculator(entrants) {
-  // seu código aqui
+  if (entrants === undefined || Object.keys(entrants).length === 0) return 0;
+  const { Adult, Senior, Child } = data.prices;
+  const { Adult: Adulto = 0, Senior: Velhinho = 0, Child: Pirralho = 0 } = entrants;
+  return ((Adulto * Adult) + (Velhinho * Senior) + (Pirralho * Child));
 };
 
 function schedule(dayName) {
-  // seu código aqui
+  const days = dayName ? [dayName] : Object.keys(data.hours);
+  return days.reduce((acc, day) => {
+    const { open, close } = data.hours[day];
+    const text = day === 'Monday'
+      ? 'CLOSED'
+      : `Open from ${open}am until ${close - 12}pm`
+    acc[day] = text;
+    return acc;
+  }, {});
 };
 
 function animalCount(species) {
-  // seu código aqui
+  if (species === undefined) {
+    const animaisCount = data.animals.reduce((emptyValue, CurrentValue) => {
+      emptyValue[`${CurrentValue.name}`] = CurrentValue.residents.length
+      return emptyValue
```

Comment:
Te sugiro também modificar alguns nomes das variáveis, porque o `emptyValue` retorna um valor, ainda mais por ele ser o acumulador do reduce, sacou? 
```suggestion
      return accumulator
```

Reply:
Nesse caso em especifico, eu tava aprendendo o reduce, então pro meu entendimento eu coloquei o primeiro paraetro como emptyValue, já que ele inicia sendo um valor vazio saca?
Eu não tive tempo pra refatorar o código